clang/lib/Sema/HLSLBuiltinTypeDeclBuilder.cpp

```diff
@@ -483,6 +532,55 @@ BuiltinTypeMethodBuilder &BuiltinTypeMethodBuilder::dereference(T Ptr) {
   return *this;
 }
 
+template <typename T>
+BuiltinTypeMethodBuilder &
+BuiltinTypeMethodBuilder::getResourceHandle(T ResourceRecord) {
```

Comment:
nit: the name of this is 'getResourceHandle' but it doesn't look like its getting a resource handle? The resource handle is pushed onto the statement list vector. Does the name make sense?